ome/resource_deploy.go

```diff
@@ -962,25 +961,58 @@ func updateDeploymentState(stateTemplateDeployment, planTemplateDeployment *mode
 
 	switch usedDeviceInput {
 	case clients.ServiceTags:
-		devSTsTfsdk.Elems = filteredDevSTList
-		stateTemplateDeployment.DeviceServicetags = devSTsTfsdk
-		stateTemplateDeployment.DeviceIDs = planTemplateDeployment.DeviceIDs
+		devSTsTfsdk, _ := types.SetValue(
+			types.StringType,
+			filteredDevSTList,
+		)
+		if !devSTsTfsdk.IsUnknown() {
+			stateTemplateDeployment.DeviceServicetags = devSTsTfsdk
+		}
+		if planTemplateDeployment.DeviceIDs.IsUnknown() {
+			stateTemplateDeployment.DeviceIDs = planTemplateDeployment.DeviceIDs
+		}
 	case clients.DeviceIDs:
-		devIDsTfsdk.Elems = filteredDevIDList
-		stateTemplateDeployment.DeviceIDs = devIDsTfsdk
-		stateTemplateDeployment.DeviceServicetags = planTemplateDeployment.DeviceServicetags
-	}
-
-	stateTemplateDeployment.BootToNetworkISO = planTemplateDeployment.BootToNetworkISO
-	stateTemplateDeployment.DeviceAttributes = planTemplateDeployment.DeviceAttributes
-	stateTemplateDeployment.OptionsContinueOnWarning = planTemplateDeployment.OptionsContinueOnWarning
-	stateTemplateDeployment.PowerStateOff = planTemplateDeployment.PowerStateOff
-	stateTemplateDeployment.OptionsPrecheckOnly = planTemplateDeployment.OptionsPrecheckOnly
-	stateTemplateDeployment.ForcedShutdown = planTemplateDeployment.ForcedShutdown
-	stateTemplateDeployment.OptionsStrictCheckingVlan = planTemplateDeployment.OptionsStrictCheckingVlan
-	stateTemplateDeployment.OptionsTimeToWaitBeforeShutdown = planTemplateDeployment.OptionsTimeToWaitBeforeShutdown
-	stateTemplateDeployment.RunLater = planTemplateDeployment.RunLater
-	stateTemplateDeployment.Cron = planTemplateDeployment.Cron
+		devIDsTfsdk, _ := types.SetValue(
+			types.Int64Type,
+			filteredDevIDList,
+		)
+		if !devIDsTfsdk.IsUnknown() {
+			stateTemplateDeployment.DeviceIDs = devIDsTfsdk
+		}
+		if !planTemplateDeployment.DeviceServicetags.IsUnknown() {
+			stateTemplateDeployment.DeviceServicetags = planTemplateDeployment.DeviceServicetags
+		}
+	}
+	if !planTemplateDeployment.BootToNetworkISO.IsUnknown() {
```

Comment:
Do we really need to check the unknowns for each and every attribute ??

Reply:
unsetting the unknown after testing the datasource and resource